From e0c44b9d6b6d6e7b55d10f5b16006bcfe504d374 Mon Sep 17 00:00:00 2001
From: Jiang Bin <bin.jiang@windriver.com>
Date: Wed, 16 Nov 2011 14:10:24 +0800
Subject: [PATCH 42/92] crypto: add user space ring in dts

Introduce "user-space-ring" flag for reserving a job ring, exclusively,
for user space.

Signed-off-by: Naveen Burmi <NaveenBurmi@freescale.com>
Integrated-by: Jiang Bin <bin.jiang@windriver.com>
---
 arch/powerpc/boot/dts/psc9131rdb.dts |   18 +++++++++++++-----
 1 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/arch/powerpc/boot/dts/psc9131rdb.dts b/arch/powerpc/boot/dts/psc9131rdb.dts
index 3256029..21b2953 100644
--- a/arch/powerpc/boot/dts/psc9131rdb.dts
+++ b/arch/powerpc/boot/dts/psc9131rdb.dts
@@ -416,7 +416,7 @@
 			};
 
 			crypto: crypto@30000 {
-				compatible = "fsl,p4080-sec4.0", "fsl,sec4.0";
+				compatible = "fsl,p4080-sec4.0", "fsl,sec-v4.0";
 				#address-cells = <1>;
 				#size-cells = <1>;
 				reg = <0x30000 0x10000>;
@@ -425,15 +425,21 @@
 				status = "ok";
 
 				sec_jr0: jr@1000 {
-					compatible = "fsl,sec4.0-job-ring",
+					compatible = "fsl,sec-v4.0-job-ring",
 						"fsl,p4080-sec4.0-job-ring";
 					reg = <0x1000 0x1000>;
 					interrupts = <58 2>;
 					interrupt-parent = <&mpic>;
+					/* By default a Job Ring is dedicated to kernel space SEC dirver.
+					   Reserving a Job Ring, exclusively, for user space SEC driver
+					   a flag "user-space-ring" must be specified in the corresponding
+					   Job Ring node.
+					*/
+					user-space-ring;
 				};
 
 				sec_jr1: jr@2000 {
-					compatible = "fsl,sec4.0-job-ring",
+					compatible = "fsl,sec-v4.0-job-ring",
 						"fsl,p4080-sec4.0-job-ring";
 					reg = <0x2000 0x1000>;
 					interrupts = <59 2>;
@@ -441,7 +447,7 @@
 				};
 
 				sec_jr2: jr@3000 {
-					compatible = "fsl,sec4.0-job-ring",
+					compatible = "fsl,sec-v4.0-job-ring",
 						"fsl,p4080-sec4.0-job-ring";
 					reg = <0x3000 0x1000>;
 					interrupts = <60 2>;
@@ -449,11 +455,13 @@
 				};
 
 				sec_jr3: jr@4000 {
-					compatible = "fsl,sec4.0-job-ring",
+					compatible = "fsl,sec-v4.0-job-ring",
 						"fsl,p4080-sec4.0-job-ring";
 					reg = <0x4000 0x1000>;
 					interrupts = <61 2>;
 					interrupt-parent = <&mpic>;
+					/* Reserving Job Ring, exclusively, for user space SEC driver. */
+					user-space-ring;
 				};
 			};
 
-- 
1.7.0

